File: Leetcode/Leetcode-DSA-course/Arrays and Strings/Prefix Sum/3-2090 K Radius Subarray Averages/main.java
class Solution {
    public int[] getAverages(int[] nums, int k) {
        if (k == 0)
            return nums;
        int n = nums.length;
        int windowSize = 2*k +1;
        int avgs[] = new int[n];
        Arrays.fill(avgs,-1);

        if(windowSize > n) return avgs;
        long windowSum = 0;
        for(int i=0;i<windowSize;i++)
            windowSum += nums[i];
        
        avgs[k] = (int)(windowSum/windowSize);

        for(int i=windowSize;i<n;i++){
            windowSum += nums[i] - nums[i-windowSize];
            avgs[i-k] = (int)(windowSum/windowSize);
        }

        return avgs;
    }
}

File: Leetcode/Leetcode-DSA-course/Arrays and Strings/Prefix Sum/3-2090 K Radius Subarray Averages/main.py
class Solution:
    def getAverages(self, nums: List[int], k: int) -> List[int]:
        averages = [-1] * len(nums)
        if k == 0:
            return nums
        windowSize = 2 * k + 1
        n = len(nums)
        if windowSize > n:
            return averages
        
        windowSum = sum(nums[:windowSize])
        averages[k] = windowSum // windowSize

        for i in range(windowSize,n):
            windowSum += nums[i] - nums[i-windowSize]
            averages[i-k] = windowSum // windowSize

        return averages
